packages/relay/src/lib/eth.ts

```diff
@@ -509,6 +509,10 @@ export class EthImpl implements Eth {
         // Handle Contract Call or Contract Create
         gas = this.defaultGas;
       }
+
+      if (e instanceof JsonRpcError) {
```

Comment:
Move to first line and adjust to something like this. I think it should cover both scenarios where we shouldn't fallback. Basically throw if we get a valid revert message from Mirror Node
```suggestion
      if (e instanceof JsonRpcError && e.mesage !== '') {
```

Reply:
Done.